src/lvm.cpp

```diff
@@ -0,0 +1,451 @@
+                    vgSizeStr.end());
+                vgFreeStr.erase(
+                    std::remove(vgFreeStr.begin(), vgFreeStr.end(), 'G'),
+                    vgFreeStr.end());
+
+                // Convert vg_size and vg_free from string to double
+                double vgSizeGB = std::stod(vgSizeStr);
+                double vgFreeGB = std::stod(vgFreeStr);
+
+                // Calculate the percentage of free space
+                double freePercentage = (vgFreeGB / vgSizeGB) * 100;
+
+                LOG_TRACE("LVM Volume Group: {}, Total Size: {} GB, Free "
+                          "Space: {} GB, Free Percentage: {:.2f}%",
+                    vgName, vgSizeGB, vgFreeGB, freePercentage);
+
+                // Check if the free space is less than 50%
+                if (freePercentage < 50.0) {
+                    LOG_WARN("LVM Volume Group '{}' does not have enough free "
+                             "space (only {:.2f}% available).",
+                        vgName, freePercentage);
+                    allVGsHaveEnoughSpace = false;
+                }
+            }
+
+            if (!allVGsHaveEnoughSpace) {
+                throw std::runtime_error("LVM ERROR: Not all LVM volume groups "
+                                         "have at least 50% free space.");
+            }
+
+            LOG_INFO(
+                "LVM: All LVM volume groups have at least 50% free space.");
+        } catch (const std::exception& e) {
+            throw std::runtime_error(fmt::format(
+                "LVM ERROR: Failed to parse disk space information: {}",
+                e.what()));
+        }
+    } else {
+        throw std::runtime_error(
+            "LVM ERROR: Failed to check available disk space.");
+    }
+}
+
+void LVM::verifyAvailablePartitions()
+{
+    std::list<std::string> output;
+    const std::string checkPartitionsCommand
+        = "lsblk --noheadings --output MOUNTPOINT";
+    int exitCode = cloyster::runCommand(checkPartitionsCommand, output, false);
+
+    if (exitCode == 0) {
+        for (const auto& line : output) {
+            if (line == "/var") {
+                m_hasVarPartition = true;
+            } else if (line == "/opt") {
+                m_hasOptPartition = true;
+            } else if (line == "/home") {
+                m_hasHomePartition = true;
+            }
+        }
+
+        if (m_hasHomePartition) {
+            LOG_TRACE(
+                "LVM: /home is in a separate partition. We can proceed with "
+                "LVM snapshot.");
+        } else {
+            LOG_WARN("LVM: /home is not a separate partition. This may lead "
+                     "to issues when rolling back the snapshot.");
+        }
+    } else {
+        throw std::runtime_error(
+            "LVM ERROR: Failed to list available partitions.");
+    }
+}
+void LVM::checkLVMAvailability()
+{
+    LOG_INFO("LVM: Begin of availability check.")
+    verifyBootIsNotLVM();
+    checkUEFIMode();
```

Comment:
This is out of context. UEFI isn't required for LVM.

Reply:
It's 2AM and I'm on the phone now, but at least this one I can answer. 

You asked for this verification on your list. _"Verificar se o sistema foi instalado em modo UEFI (jogar um warning se não for)"_

Reply:
> It's 2AM and I'm on the phone now, but at least this one I can answer.
> 
> You asked for this verification on your list. _"Verificar se o sistema foi instalado em modo UEFI (jogar um warning se não for)"_

Yeah but it's not in this context.

Reply:
It shouldn't have been put in the middle of the LVM requirements then..